design-proposals/local-storage-migration/local-storage-migration.md

```diff
@@ -0,0 +1,261 @@
+# Overview
+
+This proposal enhances KubeVirt by adding the support for migrating VMs with
+non-shared storage.
+
+## Motivation
+
+Shared storage is today a strong requirement for migrating Virtual Machines.
+However, not all storage providers offer the ReadWriteMany access mode, therefore the VM cannot be transferred even if it only utilizes one local volume.
+
+
+KubeVirt already partially supports migrate no-shared disks like container disks
+using the libvirt feature for [migrating non-shared images](https://github.com/kubevirt/kubevirt/issues/6086). However, this doesn't apply to DataVolumes and PersistentVolumeClaims.
+
+Furthermore, the ability to migrate local storage makes a number of additional interesting features available that are not specific to local storage. A copy of the storage is made as part of the storage migration.
+Additional use-cases made possible by this mechanism include:
+  1. Switch storage class. For example, if the user wants to
+     change storage type, like in case the new storage provider provides
+     additional functionalities.
+  2. Move to a larger disk if the storage class doesn't support volume expansion
+  3. Move from container disks to PVC (this can be considered as a slightly
+     different flavor of the [Persistent containerDisk volumes](https://github.com/kubevirt/kubevirt/issues/6086))
+
+## Goals
+
+Design KubeVirt APIs for introducing non-shared storage migration.
```

Comment:
Should the goal be an api or to solve a problem? *hint*

Reply:
Well, both :)... I'll rephrase this